### DISCUSSION TITLE
How to Perform Group By with Date Extraction in Prisma

### DISCUSSION THREAD

**Discussion by Chandan-Kalita** [prisma/prisma#24169] [Q&A]
## Problem

Performing a group by operation in Prisma that involves date extraction from a timestamp column ("createdAt") in PostgreSQL is currently challenging. The goal is to sum a specific column ("totalFare") grouped by the month extracted from the "createdAt" timestamp.

## Suggested solution

I would like to be able to write a query in Prisma that performs a group by operation based on the month extracted from a timestamp column ("createdAt") and then calculates the sum of a specific column ("totalFare") for each month.

## Alternatives

To achieve the desired functionality, a workaround involves using a raw SQL query alongside Prisma's capabilities. This allows incorporating the EXTRACT function for date extraction within the GROUP BY clause.

## Additional context

Currently, Prisma's groupBy method does not directly support the EXTRACT function for date extraction within the groupBy clause. Utilizing raw SQL queries introduces the risk of SQL injection, so caution must be exercised when implementing this approach. The ability to handle date extraction within Prisma's groupBy method would streamline query operations and enhance compatibility with PostgreSQL-specific functionalities.

**Comment by ludralph** [comment]
Hi @Chandan-Kalita 👋 

Thank you for raising this question.

At the moment, using [raw queries](https://www.prisma.io/docs/orm/prisma-client/queries/raw-database-access/raw-queries) will be the way to implement this as we do not support `EXTRACT` function in Prisma queries.  There is an open [issue](https://github.com/prisma/prisma/issues/6653) that hopes to address this. Do leave a 👍 on the open issue and add your use case as well.

**Reply by Chandan-Kalita** [reply]
Ya I am currently using raw queries to solve this, but we expecting this in prisma in future, 
Do you guys have any plan regarding these db functions prisma implementation ?

**Reply by ludralph** [reply]
Hi @Chandan-Kalita 

I don't have a concrete ETA for this but i can share with you how we are thinking about these issues [here](https://github.com/prisma/prisma/discussions/25772).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
